src/compiler/emitter.ts

```diff
@@ -55,6 +55,20 @@ module ts {
     }
 
     let indentStrings: string[] = ["", "    "];
+
+    // When emitting for ES3, property access via dot notation (i.e. obj.prop ) cannot be used
+    // if the property name is a reserved word.  Use index notation (i.e. obj["prop"]).
+    let ES3ReservedWords = [

```

Comment:
I'd prefer us to just break up the SyntaxKind range to have the reserved words in a single range.  Then you can do this with just a SyntaxKind check.